### PR TITLE
Updated upgrade and migration guide for 1.9

### PR DESCRIPTION
**Pull request by stuclem**
@reasonerjt can you please review and add more reviewers as appropriate? Thanks!

### CODE REVIEW

**Comment by reasonerjt** [comment]
Please make sure this is cherry picked to 1.9 branch

**Comment by stuclem** [comment]
> Please make sure this is cherry picked to 1.9 branch

Thanks @reasonerjt. @xaleeks can you please provide the 2nd approval?

File: docs/migration_guide.md [hunk 1]
```diff
@@ -1,36 +1,28 @@
...
+Since the migration might alter the database schema and the settings of `harbor.cfg`, you should **always** back up your data before any migration.
 
-**NOTE:**
+**NOTES:**
 
 - Again, you must back up your data before any data migration.
+- Since v1.8.0, the configuration of Harbor has changed to a `.yml` file. If you are upgrading from 1.7.x, the migrator will transform the configuration file from `harbor.cfg` to `harbor.yml`. The command will be a little different to perform this migration, so make sure you follow the steps below.
+- In version 1.9.0, some containers are started by `non-root`. This does not pose problems if you are upgrading an officially released version of Harbor, but if you have deployed a customized instance of Harbor, you might encounter permission issues.
+- In previous releases, user roles took precedence over group roles in a project. In this version, if a user roles and group roles are combined so that the user has both the user and group role. This might cause the roles of certain users to change during upgrade.
```
Comment (xaleeks) [inline docs/migration_guide.md]: typo, get rid of 'if' in 'if a user roles ..'

File: docs/migration_guide.md [hunk 2]
```diff
@@ -42,28 +34,25 @@ you follow the steps below.
 3. Get the latest Harbor release package from Github:
    https://github.com/goharbor/harbor/releases
 
-4. Before upgrading Harbor, perform migration first.  The migration tool is delivered as a docker image, so you should pull the image from docker hub. Replace [tag] with the release version of Harbor (e.g. v1.5.0) in the below command:
+4. Before upgrading Harbor, perform migration first.  The migration tool is delivered as a docker image, so you should pull the image from docker hub. Replace [tag] with the release version of Harbor (for example, v1.9.0) in the command below:
```
Comment (xaleeks) [inline docs/migration_guide.md]: perform a migration first

File: docs/migration_guide.md [hunk 3]
```diff
@@ -42,28 +34,25 @@ you follow the steps below.
...
+    **NOTE:** The schema upgrade and data migration of the database is performed by core when Harbor starts, if the migration fails, please check the log of core to debug.
 
+6. Under the directory `./harbor`, run the `./install.sh` script to install the new Harbor instance. If you choose to install Harbor with components such as Notary, Clair, and chartmuseum, refer to [Installation & Configuration Guide](../docs/installation_guide.md) for more information.
 
-### Roll back from an upgrade
-For any reason, if you want to roll back to the previous version of Harbor, follow the below steps:
 
-**NOTE:** Roll back doesn't support upgrade across v1.5.0, like from v1.2.0 to v1.7.0. This is because Harbor changes DB to PostgreSQL from v1.7.0, the migrator cannot roll back data to MariaDB.    
+## Roll Back from an Upgrade
+If, for any reason, you want to roll back to the previous version of Harbor, follow the below steps:
```
Comment (xaleeks) [inline docs/migration_guide.md]: maybe communicate that we can rollback during upgrade to the state before upgrade started, but harbor doesn't support downgrades in general

File: docs/migration_guide.md [hunk 4]
```diff
@@ -1,36 +1,28 @@
-# Harbor upgrade and migration guide
+# Harbor Upgrade and Migration Guide
 
-This guide only covers upgrade and migration to version >= v1.8.0
+This guide covers upgrade and migration to versions >= v1.9.0. This guide only covers migration from v1.7.x and later to the current version. If you are upgrading from an earlier version, refer to the migration guide in the `release-1.7.0` branch to upgrade to v1.7.x first, then follow this guide to perform the migration to this version.
```
Comment (xaleeks) [inline docs/migration_guide.md]: It's migration to versions v1.9.x correct?  1.10 might introduce different behavior, we don't know for sure yet.

File: docs/migration_guide.md (hunk 1 above)
Comment (xaleeks) [inline docs/migration_guide.md]: maybe it's more accurate to say that the user has the higher permission set of the two.  I believe this is true even if user rights are of higher permissions than group rights

File: docs/migration_guide.md (hunk 4 above)
Comment (xaleeks) [reply docs/migration_guide.md]: sounds good

**Comment by xaleeks** [comment]
LGTM

**Pull request merged**